### PR TITLE
feat: adds the CourseEnrollmentSiteFilterRequested filter

### PR DESCRIPTION
**Pull request by JuanDavidBuitrago**
**Description:** 

This adds the ``CourseEnrollmentSiteFilterRequested`` filter which receive a Query Set context with enrollments information to allowing the filter pipeline to return the information that we need to filter.

An example of use is in [eox-tenant](https://github.com/eduNEXT/eox-tenant) custom filter step [pipeline](https://github.com/eduNEXT/eox-tenant/blob/master/eox_tenant/filters/pipeline.py), where it takes the enrollments information and is filtered by tenant request, returning only the enrollments in the current site for the user.

**JIRA:** N/A

**Dependencies:**

- https://github.com/openedx/edx-platform/pull/31331

**Merge deadline:** N/A

**Installation instructions:** 

Be sure to use ``edx-platform`` branch with changes https://github.com/openedx/edx-platform/pull/31331

**Testing instructions:**

The change can be tested with testing instructions in [eox-tenant plugin](https://github.com/eduNEXT/eox-tenant/pull/156).

**Reviewers:**
- [ ] @mariajgrimaldi 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** N/A

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @JuanDavidBuitrago!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by JuanDavidBuitrago** [comment]
Hi @mariajgrimaldi and @felipemontoya, could you help me review this?

File: openedx_filters/learning/filters.py [hunk 1]
```diff
@@ -442,3 +442,22 @@ def run_filter(cls, block, context):
...
+    @classmethod
+    def run_filter(cls, context):
+        """
+        Execute a filter with the signature specified.
+
+        Arguments:
+        context (QuerySet): list of all user's course enrollments
+        """
+        data = super().run_pipeline(context=context)
+        return data.get("context")
```
Comment (mariajgrimaldi) [inline openedx_filters/learning/filters.py:463]: You're missing the filter exceptions like the existing filters: https://github.com/openedx/openedx-filters/blob/96da9f887dfb8cf5834f562ad02493601ba263af/openedx_filters/learning/filters.py#L70

Comment (JuanDavidBuitrago) [reply openedx_filters/learning/filters.py:468]: I added filter exception

**Comment by JuanDavidBuitrago** [comment]
Hi @mariajgrimaldi!!! 

Could you check again the last changes? Thanks!! :smiley:

File: openedx_filters/learning/filters.py [hunk 2]
```diff
@@ -442,3 +442,27 @@ def run_filter(cls, block, context):
...
+        Custom class used to stop the course enrollment site filter process.
+        """
+
+    @classmethod
+    def run_filter(cls, context):
+        """
+        Execute a filter with the signature specified.
+
+        Arguments:
+        context (QuerySet): list of all user's course enrollments
```
Comment (mariajgrimaldi) [inline openedx_filters/learning/filters.py:465]: If this is a queryset, then why is the argument called context?

File: openedx_filters/learning/filters.py [hunk 3]
```diff
@@ -442,3 +442,27 @@ def run_filter(cls, block, context):
         """
         data = super().run_pipeline(block=block, context=context)
         return data.get("block"), data.get("context")
+
+
+class CourseEnrollmentSiteFilterRequested(OpenEdxPublicFilter):
```
Comment (mariajgrimaldi) [inline openedx_filters/learning/filters.py:447]: The other filters don't have the word _filter_ on them, can we remove it from here? 
Also, see how CourseEnrollmentRequested tells us the filter is executed before the course enrollment object is created, so we can stop the process or modify the object. We aim to follow that convention with each filter, can we try that here? Using the PR from edx-platform as an example of where you're using this filter, the name could be something like CourseEnrollmentQuerysetRequested (a bit long), do you have another suggestion?

**Review by mariajgrimaldi — CHANGES_REQUESTED** [review]

File: openedx_filters/learning/filters.py (hunk 2 above)
Comment (JuanDavidBuitrago) [reply openedx_filters/learning/filters.py]: I changed this to ``enrolments``.

File: openedx_filters/learning/filters.py [hunk 4]
```diff
@@ -442,3 +442,29 @@ def run_filter(cls, block, context):
         """
         data = super().run_pipeline(block=block, context=context)
         return data.get("block"), data.get("context")
+
+
+class CourseEnrollmentQuerysetRequested(OpenEdxPublicFilter):
+    """
+    Custom class used to filter user's course enrollments by site, when a request is made by the user.
```
Comment (mariajgrimaldi) [inline openedx_filters/learning/filters.py:449]: Great description! The issue is that not many people will use this filter as described here. That type of documentation goes to the pipeline steps. I suggest something like: 
```suggestion
    Custom class used to create course enrollments queryset filters and its custom methods.
```
This following the other filters docstrings.

File: openedx_filters/learning/filters.py [hunk 5]
```diff
@@ -442,3 +442,29 @@ def run_filter(cls, block, context):
...
+    Basically it modifies the enrollments queryset.
+    """
+
+    filter_type = "org.openedx.learning.course_enrollment_queryset.requested.v1"
+
+    class PreventEnrollmentSiteFilter(OpenEdxFilterException):
```
Comment (mariajgrimaldi) [inline openedx_filters/learning/filters.py:455]: It would help if you changed the name here too

File: openedx_filters/learning/filters.py (hunk 4 above)
Comment (JuanDavidBuitrago) [reply openedx_filters/learning/filters.py]: Thanks @mariajgrimaldi, I modified this part.

File: openedx_filters/learning/filters.py (hunk 5 above)
Comment (JuanDavidBuitrago) [reply openedx_filters/learning/filters.py]: Yes, you're right. I forgot to change this.

Thanks for your comments. Could you check again?

File: openedx_filters/learning/filters.py [hunk 6]
```diff
@@ -442,3 +442,27 @@ def run_filter(cls, block, context):
...
+class CourseEnrollmentQuerysetRequested(OpenEdxPublicFilter):
+    """
+    Custom class used to create course enrollments queryset filters and its custom methods.
+    """
+
+    filter_type = "org.openedx.learning.course_enrollment_queryset.requested.v1"
+
+    class PreventCourseEnrollmentQueryset(OpenEdxFilterException):
+        """
+        Custom class used to stop the course enrollment queryset filter process.
```
Comment (mariajgrimaldi) [inline openedx_filters/learning/filters.py:456]: `Custom class used to stop the course enrollment queryset request process.`

Comment (mariajgrimaldi) [inline openedx_filters/learning/filters.py:456]:
```suggestion
        Custom class used to stop the course enrollment queryset request process.
```

File: openedx_filters/learning/filters.py [hunk 7]
```diff
@@ -442,3 +442,27 @@ def run_filter(cls, block, context):
...
+
+
+class CourseEnrollmentQuerysetRequested(OpenEdxPublicFilter):
+    """
+    Custom class used to create course enrollments queryset filters and its custom methods.
+    """
+
+    filter_type = "org.openedx.learning.course_enrollment_queryset.requested.v1"
+
+    class PreventCourseEnrollmentQueryset(OpenEdxFilterException):
```
Comment (mariajgrimaldi) [inline openedx_filters/learning/filters.py:454]: The wording here could be clearer. I like better: PreventEnrollmentQuerysetRequest, what do you folks think?

Comment (JuanDavidBuitrago) [reply openedx_filters/learning/filters.py]: It's ok, I agree with that. Let me know if you consider another change.

File: openedx_filters/learning/tests/test_filters.py [hunk 8]
```diff
@@ -264,6 +266,19 @@ def test_halt_enrollment_process(self, enrollment_exception, attributes):
...
+
+        Expected behavior:
+            - The filter should return the enrollments to orgs.
+        """
+        expected_enrollments = Mock()
+
+        enrollments = CourseEnrollmentQuerysetRequested.run_filter(expected_enrollments)
+
+        self.assertEqual(expected_enrollments, enrollments)
+
```
Comment (mariajgrimaldi) [inline openedx_filters/learning/tests/test_filters.py:281]: I think you're missing [this kind](https://github.com/openedx/openedx-filters/blob/main/openedx_filters/learning/tests/test_filters.py#L256) of tests. Could you add them please?

Comment (JuanDavidBuitrago) [reply openedx_filters/learning/tests/test_filters.py:281]: Thanks!! I added the test.

**Review by mariajgrimaldi — APPROVED** [review]
LGTM

**Comment by openedx-webhooks** [comment]
@JuanDavidBuitrago 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-filters&entry.867055334=https://github.com/openedx/openedx-filters/pull/47&entry.1484655318=https://github.com/JuanDavidBuitrago&entry.752974735=2022-11-16+12:28&entry.1917517419=2023-01-18+15:02&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**